helix-view/src/clipboard.rs

```diff
@@ -0,0 +1,193 @@
+// Implementation reference: https://github.com/neovim/neovim/blob/f2906a4669a2eef6d7bf86a29648793d63c98949/runtime/autoload/provider/clipboard.vim#L68-L152
+
+use anyhow::Result;
+use std::borrow::Cow;
+
+pub trait ClipboardProvider: std::fmt::Debug {
```

Comment:
I guess this trait gives us the option to introduce more providers later, is that the intent here? Otherwise we could also simply use `Option<CommandProvider>` and avoid the `Box<dyn T>`

Reply:
Yeah that was the intent, but I agree this could as well be `Option<CommandProvider>` right now.